File: eir-front-angular/src/app/services/dtos/api-exception.ts
import {isNull} from '../../helpers';

export interface ApiException {
  apiErrorCode: string;
  details: string;
  stack: string;
}

export const SESSION_NOT_FOUND_CODE = 'SES001';

export function isError<T>(response: T | ApiException) {
  const parsedResponse: ApiException = response as ApiException;
  return !isNull(parsedResponse) && !isNull(parsedResponse.apiErrorCode);
}

File: eir-back/src/main/kotlin/com/eir/gdr/ApiException.kt
package com.eir.gdr

data class ApiException(
    var apiErrorCode: String? = null,
    var details: String? = null,
    var stack: String? = null
) : Exception(details) {
    companion object {
        val sessionNotFound = ApiException("SES001", "Session not found")
        val insufficientExperience = ApiException("EXP001", "Esperienza insufficiente per aumentare la scheda.")
    }
}
